lambdas/src/services/auth-request-validator.ts

```diff
@@ -1,32 +1,30 @@
 import { APIGatewayProxyEventQueryStringParameters } from "aws-lambda/trigger/api-gateway-proxy";
-import { ValidationResult } from "../types/validation-result";
+import { SessionValidationError } from "../types/errors";
 
 export class AuthorizationRequestValidator {
     validate(
         queryStringParams: APIGatewayProxyEventQueryStringParameters | null,
         sessionClientId: string,
         expectedRedirectUri: string,
-    ): ValidationResult {
+    ) {
         if (!queryStringParams) {
-            return { isValid: false, errorMsg: "Missing querystring parameters" };
+            throw new SessionValidationError("Session Validation Exception", "Missing querystring parameters");
         }
 
         const clientId = queryStringParams["client_id"];
         const redirectUri = queryStringParams["redirect_uri"];
         const responseType = queryStringParams["response_type"];
-        let errorMsg = null;
         if (!clientId) {
-            errorMsg = "Missing client_id parameter";
+            throw new SessionValidationError("Session Validation Exception", "Missing client_id parameter");
```

Comment:
Just a question: For the end user experience, it might be better to throw a complex error containing all the validation failures rather than them having to call the api multiple times? Less requests this way and not limiting the error to first reported error?

Reply:
For now the spirit is a like for like interpretation of the way it is done in Java land and not to really to extend it in anyway for now, your suggestion is valid could be for future consideration.